IntelFsp2Pkg/Include/FspEas/FspApi.h

```diff
@@ -77,9 +161,40 @@ typedef struct {
   /// Current boot mode.
   ///
   UINT32                      BootMode;
-  UINT8                       Reserved1[8];
+  ///
+  /// Optional event handler for the bootloader to be informed of events occurring during FSP execution.
+  /// This value is only valid if Revision is >= 2.
+  ///
+  UINT32                      FspEventHandler;
+  UINT8                       Reserved1[4];
 } FSPM_ARCH_UPD;
 
+typedef struct {
+  ///
+  /// Revision Revision of the structure is 1 for this version of the specification.
+  ///
+  UINT8                      Revision;
+  UINT8                      Reserved[3];
+  ///
+  /// Length Length of the structure in bytes. The current value for this field is 32.
+  ///
+  UINT32                      Length;
+  ///
+  /// FspEventHandler Optional event handler for the bootloader to be informed of events
+  /// occurring during FSP execution.
+  ///
+  FSP_EVENT_HANDLER           FspEventHandler;

```

Comment:
This needs to be UINT32 too.

Reply:
I was not sure, in the EDK2 file it is not a pointer either, does it need to change to UINT32 still?

https://github.com/tianocore/edk2/blob/master/IntelFsp2Pkg/Include/FspEas/FspApi.h#L186

I did change the one on line 168 to UINT32 because it was a pointer.

Reply:
I updated it now, I see now since the compiler will also treat function pointers the same as other pointers it could cause issue to allocate 32 or 64 bits.